### PR TITLE
changed GET method to SESSION as to prevent failure during export of …

### PR DESCRIPTION
**Pull request by naphets123**
…large amounts of data

### CODE REVIEW

**Comment by naphets123** [comment]
https://trello.com/c/AEh5qwmR/384-error-on-export-of-large-amounts-of-data-due-to-get-request

File: include/people_export.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
     '
 	SELECT p.*, DATE_FORMAT(FROM_DAYS(DATEDIFF(NOW(), p.date_of_birth)), "%Y")+0 AS age
 	FROM people AS p
-	WHERE camp_id = :campid AND (NOT deleted OR deleted IS NULL) '.($_GET['ids'] ? 'AND p.id in ('.$_GET['ids'].')' : ' AND FALSE'),
+	WHERE camp_id = :campid AND (NOT deleted OR deleted IS NULL) '.($_SESSION['export_ids'] ? 'AND p.id in ('.$_SESSION['export_ids'].')' : ' AND FALSE'),
```
Comment (HaGuesto) [inline include/people_export.php]: could you try to passing the exportids like the campid?

File: include/stock_export.php [hunk 2]
```diff
@@ -7,7 +7,7 @@
     LEFT OUTER JOIN locations AS l ON l.id = boxes.location_id
     LEFT OUTER JOIN genders AS g ON g.id = p.gender_id
     LEFT OUTER JOIN sizes AS s ON s.id = boxes.size_id 
-    WHERE l.camp_id = :campid AND (NOT boxes.deleted OR boxes.deleted IS NULL) '.($_GET['ids'] ? 'AND boxes.id in ('.$_GET['ids'].')' : ' AND FALSE'),
+    WHERE l.camp_id = :campid AND (NOT boxes.deleted OR boxes.deleted IS NULL) '.($_SESSION['export_ids'] ? 'AND boxes.id in ('.$_SESSION['export_ids'].')' : ' AND FALSE'),
```
Comment (HaGuesto) [inline include/stock_export.php]: could you try to passing the exportids like the campid?

File: include/people_export.php [hunk 3]
```diff
@@ -4,7 +4,7 @@
...
     ['campid' => $_SESSION['camp']['id']]
 );
```
Comment (HaGuesto) [inline include/people_export.php]: I would add `unset($_SESSION['export_ids'])` here.

File: include/stock_export.php [hunk 4]
```diff
@@ -7,7 +7,7 @@
...
     ['campid' => $_SESSION['camp']['id']]
 );
```
Comment (HaGuesto) [inline include/stock_export.php]: I would add `unset($_SESSION['export_ids'])` here.

**Comment by naphets123** [comment]
@HaGuesto : i see, whouldn't giving them seperate names prevent this exploit? export_boxes_ids, export_stock_ids .... we could do the unset too

File: include/people_export.php (hunk 1 above)
Comment (naphets123) [reply include/people_export.php]: do you mean by like [campid => $_SESSION['camp_id'], exportids=> $_SESSION['export_ids'] ?

Comment (HaGuesto) [reply include/people_export.php]: yes, exactly

**Comment by HaGuesto** [comment]
> @HaGuesto : i see, whouldn't giving them seperate names prevent this exploit? export_boxes_ids, export_stock_ids .... we could do the unset too

Yes, it would. Do it :)

**Comment by jamescrowley** [comment]
@HaGuesto @naphets123 apologies if this has already been discussed, but is there any way we can achieve this without relying on session data again? It just seems like we're setting ourselves up for more strange bugs, given the existing issues we have with SESSION data being used to manage state. Could we perhaps avoid the redirect and do the export immediately while we have the $_POST data? Or ensure the data is POSTed to the export endpoint in the first place?

**Comment by HaGuesto** [comment]
> @HaGuesto @naphets123 apologies if this has already been discussed, but is there any way we can achieve this without relying on session data again? It just seems like we're setting ourselves up for more strange bugs, given the existing issues we have with SESSION data being used to manage state. Could we perhaps avoid the redirect and do the export immediately while we have the $_POST data? Or ensure the data is POSTed to the export endpoint in the first place?

@naphets123 I talked to @jamescrowley and using the SESSION variable is ok and the fastest and easiest solution for now.

File: include/people_export.php (hunk 1 above)
Comment (naphets123) [reply include/people_export.php]: In this case we couldn't use the  Syntax $_SESSION['export_ids'] ? Option A : Option B
So we would add some additional checks at another point

**Comment by HaGuesto** [comment]
@naphets123 missing onw commit to escape the SQL queries.

**Comment by naphets123** [comment]
Done

File: include/products_export.php [hunk 5]
```diff
@@ -2,17 +2,25 @@
...
+
 $result = db_query(
     'SELECT p.*, sg.label AS sizegroup, g.label AS gender, p.value AS drops, COALESCE(SUM(s.items),0) AS items, pc.label AS category 
 		FROM products AS p
 		LEFT OUTER JOIN genders AS g ON g.id = p.gender_id
 		LEFT OUTER JOIN sizegroup AS sg ON sg.id = p.sizegroup_id
 		LEFT OUTER JOIN stock AS s ON s.product_id = p.id AND NOT s.deleted '.($locations ? ' AND s.location_id IN ('.$locations.')' : '').'
 		LEFT OUTER JOIN product_categories AS pc ON p.category_id = pc.id
-		WHERE (NOT p.deleted OR p.deleted IS NULL) AND camp_id = :camp_id '.($_GET['ids'] ? 'AND p.id IN ('.$_GET['ids'].')' : 'AND FALSE').'
+		WHERE (NOT p.deleted OR p.deleted IS NULL) AND camp_id = ? '.($_SESSION['export_ids_products'] ? 'AND p.id IN ('.$id_pars.')' : 'AND FALSE').'
```
Comment (HaGuesto) [inline include/products_export.php]: wicked :)

Does this mean that a list of ids cannot be passed as a variable to db_query, e.g.
```
camp_id = :camp_id AND p.id IN :id_list
```

Could you send me a link to the doc where you found that solution?

Comment (HaGuesto) [reply include/products_export.php]: https://phpdelusions.net/pdo#in

**Pull request merged**